src/bin/server/recent_transactions.rs

```diff
@@ -0,0 +1,113 @@
+use std::collections::VecDeque;
+
+use at2_node::{FullTransaction, ThinTransaction};
+use drop::crypto::sign;
+use tokio::sync::{mpsc, oneshot};
+
+const LATEST_TRANSACTIONS_MAX_SIZE: usize = 10;
+
+#[derive(snafu::Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("gone on send"))]
+    GoneOnSend,
+    #[snafu(display("gone on recv"))]
+    GoneOnRecv,
+}
+
+#[derive(Debug)]
+enum Commands {
+    Put {
+        thin: ThinTransaction,
+        sender: Box<sign::PublicKey>,
+        resp: oneshot::Sender<()>,
+    },
+    GetAll {
+        resp: oneshot::Sender<Vec<FullTransaction>>,
+    },
+}
+
+#[derive(Clone)]
+pub struct RecentTransactions {
+    agent: mpsc::Sender<Commands>,
+}
+
+/// Own the recent transactions
+struct RecentTransactionsHandler(VecDeque<FullTransaction>);
+
+impl RecentTransactions {
+    pub fn new() -> Self {
+        Self {
+            agent: RecentTransactionsHandler::new().spawn(),
+        }
+    }
+
+    /// Add a new transaction
+    pub async fn put(
+        &self,
+        sender: Box<sign::PublicKey>,
+        thin: ThinTransaction,
+    ) -> Result<(), Error> {
+        let (tx, rx) = oneshot::channel();
+
+        self.agent
+            .send(Commands::Put {
+                sender,
+                thin,
+                resp: tx,
+            })
+            .await
+            .map_err(|_| Error::GoneOnSend)?;
+
+        rx.await.map_err(|_| Error::GoneOnRecv)
```

Comment:
Is that some kind of `await` with channels? If yes, can you replace this with a 'real' `async`?